ipfx/nwb_writer.py

```diff
@@ -0,0 +1,71 @@
+import pynwb
+from allensdk.core.nwb_data_set import NwbDataSet
+import ipfx.nwb_reader as nwb_reader
+
+
+class NwbWriter(object):
+
+    def __init__(self, nwb_file_name):
+        self.nwb_file_name = nwb_file_name
+
+    def add_spike_times(self, sweep_num, spike_times):
+        raise NotImplementedError
+
+    def save(self):
+        raise NotImplementedError
+
+
+class Nwb1Writer(NwbWriter):
+
+    def __init__(self, nwb_file_name):
+        NwbWriter.__init__(self, nwb_file_name)
+        self.nwbfile = NwbDataSet(self.nwb_file_name)
+
+    def add_spike_times(self, sweep_num, spike_times):
+        self.nwbfile.set_spike_times(sweep_num, spike_times)
+
+    def save(self):
+        pass
+
+
+class Nwb2Writer(NwbWriter):
+
+    def __init__(self, nwb_file_name):
+        NwbWriter.__init__(self, nwb_file_name)
+
+        io = pynwb.NWBHDF5IO(self.nwb_file_name, 'a')
+        self.nwbfile = io.read()
+        io.close()
+
+        self.nwbfile.add_unit_column('sweep_number', description="sweep number")
```

Comment:
what happens if there is already a sweep_number column?

Reply:
It will error out -- good point. Also this brings me to needing to check if Units table already exists and remove it. Do you know what is the way to delete a container?

Reply:
why do you need to replace the table of units? Surely you just want to overwrite (or error out) one line?

Reply:
I changed implementation to using processing module and timeseries. Units table turned out to be a poor fit because it assumes a all spikes for unit saved in a single table row. In my case I need to save row for each sweep for the same unit. I also tried using DynamicTable, but had no luck with saving spike times when sweeps had different number of spikes. @NileGraddis using timeseries was your original suggestion. :) Hope you still like it.

Reply:
I have no problem with this implementation (since it seems like there is no official way to resolve this issue).

Reply:
This seems to obviate the original comment, but still ... what happens if there are already some spike times in this file?

Reply:
If time series for the sweep already exists, it will error out.